plugins/module_utils/module_helper.py

```diff
@@ -133,10 +152,60 @@ def text(self):
         return self.msg or str(self.exc_val)
 
 
+class NamedDeprecation(object):
```

Comment:
```suggestion
class _NamedDeprecation(object):
```
To make sure this is not instantiated directly by users (and thus avoiding the parameter check).

Reply:
I can do it, of course, but to me it sounds a bit like security through obscurity. If users really want to go around the check, there's no stopping them.

Reply:
Done, coming up in the next push.